src/Client/Connection.hpp

```diff
@@ -124,7 +124,7 @@ template<class BUFFER, class NetProvider>
 ConnectionImpl<BUFFER, NetProvider>::~ConnectionImpl()
 {
 	assert(refs == 0);
-	if (!strm.has_status(SS_DEAD)) {
+	if (strm.get_fd() != -1) {
```

Comment:
Nit: We rely here that closed connection always had `fd` equal to `-1`. We should describe it in `UnixStream` class (to fixate this behavior) or introduce new `stem.is_closed()` method.